src/main/scala/viper/gobra/util/PluginAwareChopper.scala

```diff
@@ -0,0 +1,54 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2025 ETH Zurich.
+
+package viper.gobra.util
+
+import viper.silver.ast.utility.chopper.{ChopperLike, Cut, Edges, SCC, Vertices, ViperGraphs}
+
+// Chopper with computation of dependencies extended to support the termination plugin and other plugins
+// used by Gobra.
+object PluginAwareChopper extends ChopperLike with ViperGraphs with PluginAwareEdges with Vertices with Cut with SCC
+
+trait PluginAwareEdges extends Edges {
```

Comment:
```suggestion
/**
  * Extends the chopper's dependency computation to add artificial dependencies for domains ending in 'WellFoundedOrder',
  * their axioms, and subexpressions.
  * These artificial dependencies ensure that these axioms are retained by the chopper, which is necessary
  * to ensure successful verification of the chopped program because these axioms define well-founded orders
  * via "bounded" and "decreasing" functions. Without these artificial dependencies, the chopper would remove
  * these axioms as the decreases measures do not explicitly depend on, i.e., call these domain functions.
  */
trait PluginAwareEdges extends Edges {
```